src/devices/usbCamera/matlabscripts/FPS_check.m

```diff
@@ -0,0 +1,47 @@
+%read file from yarpdatadumber format see above
+%
+%yarpdatadumper --name /log --rxTime --txTime --type image
+%yarp connect /grabber /log fast_tcp
+%
+
+[FileName,PathName,FilterIndex] = uigetfile('.log')
+out=horzcat(PathName,FileName);
+allTable=readtable(out);
+
+%calculate
+info=allTable(100:end-100,1:3);%remove first and last 100 frames
+info=info{:,:};
+infoshifted=zeros(size(info));
+infoshifted(2:end,:)=info(1:end-1,:);
+timediff=info-infoshifted;%time between two frames
+FPS=1./(timediff);
+FPS=FPS(2:end,2:2);
+
+%mean
+m=mean(FPS);
+s=std(FPS);
+
+%show all plots
+tiledlayout(3,1)
+nexttile
+histogram(FPS);
+xlabel('FPS');
+ylabel('Frame number');
+title('Frame distribution')
+grid on
+
+nexttile
+mytime=info(2:end,2:2)-info(2,2);
+plot(mytime,FPS(:,:));
+xlabel('Time');
+ylabel('FPS');
+title('FPS in time')
+grid on
+
+nexttile
+mytime=info(2:end,2:2)-info(2,2);
+plot(mytime,timediff(2:end,2:2));
+xlabel('Time');
+ylabel('Frame interval in sec');
+title('Frame interval in time')
+grid on
```

Comment:
What is the purpose of this script? Usually the scripts are put in the `script` folder and installed with the `module`/`device`, but if it is just a script that you used for the development maybe it is not worth to add to the repository

Reply:
I will revert them

Reply:
This script is used to check if the output from the usbcamera is correct. We can call it a system test. In the future we will need more of these test for checking other specific problems on the output. It could be useful also for the customer, we can think to install it.
In my opinion, should be near code, otherway it is difficult to find/execute it.